src/api/libcellml/issue.h

```diff
@@ -100,6 +100,7 @@ class LIBCELLML_EXPORT Issue
      */
     enum class Cause
     {
+        ANALYSER,
```

Comment:
I was of two minds about this ... most of the things in this enum are entities, which is how it should be.  The Cause of an error found in the analyser isn't the Analyser itself, it's something the Analyser has found within an entity?  I wasn't quite comfortable with the Generator being on this list either ... ho hum ...

Reply:
Yeah, but... maybe something for another PR...? :)

Reply:
Ehhh ... global search and delete for `->setCause(Issue::Cause::ANALYSER)` and you're done ...

Reply:
Sure, but... what should be done instead? I mean that if I remove calls to `setCause(Issue::Cause::ANALYSER)` then the issue will be considered to be of the `UNDEFINED` cause. Are we ok with that?

Anyway, I am going to do it and then we can refine things further in another issue, if needed. Will therefore remove `ANALYSER` from `Issue::Cause` (as well as `GENERATOR` which is not used anymore!), no problems.

Done in commit 395a18eb.

Reply:
The "cause" should be the entity associated with the issue?  If it's a problem with a variable, then VARIABLE etc ...? (ps, only unresolving so you see this comment ;) )

Reply:
Oh, ok, I see what you mean now. Makes sense actually. End of the week... :)